docs/containers-registries.conf.5.md

```diff
@@ -254,6 +259,28 @@ in order, and use the first one that exists.
 
 Note that a mirror is associated only with the current `[[registry]]` TOML table. If using the example above, pulling the image `registry.com/image:latest` will hence only reach out to `mirror.registry.com`, and the mirrors associated with `example.com/foo` will not be considered.
 
+### Enabling Additional Layer Store to access to private registries
+
+The `additional-layer-store-auth-helper` option enables to pass registry
+credentials to Additional Layer Store so that it can access private registries.
+
+Additional Layer Store needs to provide a helper binary. This helper binary is
+registered to `additional-layer-store-auth-helper` option. Every time pulling of
+an image happens, the specified helper binary is executed and receives registry
+credentials from stdin in the following format.
+
+```json
+{
+  "image-reference": {
+    "username": "username",
+    "password": "password",
+    "identityToken": "identityToken"
+  }
```

Comment:
It’s completely unclear from this syntax which of the strings are constants and which of them are variables — especially when “image-reference” is not a constant. I don’t know what’s a good way to do that here, either avoid ``` and format explicitly using the man-page `constants` and _variables_ syntax, or maybe use `"$username"` to indicate that it is a variable?

What is the format of “image-reference”? “$repo{:$tag|@$digest}" would be a bit more explicit.

Is it, long-term, guaranteed that there will be only one “image-reference”? Or can the map contain multiple entries and should the helpers be ready to accept that? I guess the latter, otherwise the `map[]` format would not make sense.

---

Thinking about the evolution, if any, more:
- Do we need a version number? Probably not, we can define an `additional-layer-store-auth-helper2` field.
- Making the top-level object’s fields directly interpreted as image references means we can’t add any other field to the object in the future. Do we care? Not _urgently_, we can always use …`helper2` with a new format. I also can’t think of any field we might want to add, but that doesn’t mean much.

Reply:
>  or maybe use `"$username"` to indicate that it is a variable?

> What is the format of “image-reference”? “$repo{:$tag|@$digest}" would be a bit more explicit.

Thanks for the suggestions. Fixed.

> Is it, long-term, guaranteed that there will be only one “image-reference”? Or can the map contain multiple entries and should the helpers be ready to accept that? I guess the latter, otherwise the map[] format would not make sense.

There is no guarantee of "image-reference" being only one. The reason for the image reference not being a field is that I initially tried to make the struct the same format as DockerAuthConfig which is already used in the codebase. But, if needed, it's also ok for me to extend the struct to make `image-reference` one of the fields (not the key of the struct).

> version

>  add any other field to the object in the future

Currently I can’t think of any field we might want to add. Maybe we can `additional-layer-store-auth-helper2` when we need to change the format.